provider/inspec.rb

```diff
@@ -60,6 +61,32 @@ def generate_resource(data)
       generate_properties(data.clone, data.object.all_user_properties)
     end
 
+    # Generate the IAM policy for this object. This is used to query and test
+    # IAM policies separately from the resource itself
+    def generate_iam_policy(data)
```

Comment:
@rambleraptor want to give this another quick look? Refactored to include your new changes to the data object